site/content/entry/citra-progress-report-2020-q2.md

```diff
@@ -0,0 +1,699 @@
++++
+
+<!-- Note: forum should be updated -->
+
+It’s been more than a year since the last progress report, not for lack of progress, but for lack of writers.
+To fill in the gap, developers [jroweboy](https://github.com/jroweboy) and [FearlessTobi](https://github.com/FearlessTobi) independently wrote drafts for a new progress report, and another developer [zhaowenlan1779](https://github.com/zhaowenlan1779) merged their works and added more content. Together, we are able to present you with a quick update on all the changes we've had since 2019 Q1.
+
+Since it's been such a long time since the last one, you may already be very familiar with several of these features, but there are also a few here that we haven't announced until now!
+
+Also, if you don't know already, Citra Android has already been released! Read this blog post (todo: link) if you are interested.
+
+# Contents
+
+1. [New Features]({{< relref "#new-features" >}})
+1. [Emulation Accuracy]({{< relref "#emulation-accuracy" >}})
+1. [Frontend Improvements]({{< relref "#frontend-improvements" >}})
+1. [Related to Citra]({{< relref "#related-to-citra" >}})
+1. [Conclusion]({{< relref "#conclusion" >}})
+
+# New Features
+
+## Save States ([#4908](https://github.com/citra-emu/citra/pull/4908), [#5223](https://github.com/citra-emu/citra/pull/5223), [#5256](https://github.com/citra-emu/citra/pull/5256)) by [hamish-milne](https://github.com/hamish-milne) and others
+
+Adding save states to Citra is a tremendous undertaking that developers shuddered to think about.
+In a modern emulator like Citra, games don't directly access the hardware anymore, but instead communicate with the 3DS operating system.
+This 3DS operating system communicates with the 3DS kernel, and the kernel will manage all of the very low level communication with the hardware.
+
+In retro console emulators, save states usually only need to store and restore the current state of all of the hardware, but in Citra, we have to do so much more.
+Every last operating service and every single kernel object need to be serialized in addition to the hardware state.
+
+In total, Hamish edited over 500 files with these changes and there was still more work to do. [zhaowenlan1779](https://github.com/zhaowenlan1779) helped make a simple UI and the file format for save states. Several developers including [lioncash](https://github.com/lioncash), [zhaowenlan1779](https://github.com/zhaowenlan1779), [B3n30](https://github.com/B3n30), [BreadFish64](https://github.com/BreadFish64) and [vitor-k](https://github.com/vitor-k) also helped by reviewing the code and pointing out errors. Together, we finally got this huge PR to a mergeable state.
+
+This is why developers have long feared working on save states, it's a thankless job where you must dig through every last bit of code in Citra, and decide if it needs to be added to the save state or not, and if you get anything wrong, well, then the game just simply breaks in unexpected ways.
```

Comment:
For the first change, I don't have a strong opinon. For the second, I think the current version reads better (personal opinion). For the third change, while this section wasn't written by me, I have indeed checked its word choice (since this was a new word to me :p), and I believe it's correct

> thankless: (of a job or task) difficult or unpleasant and not likely to bring one pleasure or the appreciation of others.

Reply:
> 
> 
> For the first change, I don't have a strong opinon. For the second, I think the current version reads better (personal opinion). For the third change, while this section wasn't written by me, I have indeed checked its word choice (since this was a new word to me :p), and I believe it's correct
> 
> > thankless: (of a job or task) difficult or unpleasant and not likely to bring one pleasure or the appreciation of others.

I just thought that ungrateful sounds better but they both have nearly the same meaning.